### ISSUE TITLE
How to set resource-id/content-desc in flutter app for automation using Appium?

### ISSUE THREAD

**Issue by maulikpatel890** [flutter/flutter#67037]
To automate flutter app using Appium, we need to set the resource-id (for Android) and accessibility-Id/content-desc (for iOS) of any element/control in the flutter app. To get resource-id and accessibility-Id in the Appium tree view of flutter app, what should be done? Please help with possible solutions to achieve this.

Please refer below screenshot of Appium tree view which shows the missing resource-id.

[![enter image description here][1]][1]

  [1]: https://i.stack.imgur.com/OVcyB.png

Flutter version = 1.17.5

Please take a note that we have already tried "key" property of Widget for unique identification but it didn't work.

**Comment by darshankawar** [comment]
@maulikpatel890 Appium is 3rd party tool for Flutter. You may check out below plugins and repos to get answers on your question.

https://pub.dev/packages/appium_driver

https://github.com/truongsinh/appium-flutter-driver

Also, Please see https://flutter.dev/community for resources, you may also get some help if you post it on Stack Overflow and if you need help with your code, please see https://www.reddit.com/r/flutterhelp/
Closing, as this isn't an issue with Flutter itself. If you disagree, please write in the comments and I will reopen it.
Thank you

**Comment by maulikpatel890** [comment]
Thanks @darshankawar  for the quick reply. 

To help you better understand the issue, I am using traditional Appiumdriver (AndroidDriver/IOSdriver) for flutter app automation with UiAutomator2/XCUITest. (I am not using flutterdriver explicitly here.) This Automator just reads the whatever key set for an element in the app.  
In any native apps, we get the resource-id/accessibility-Id/content-desc based on the value set in the application layout/widget. But, the same is not achieved in flutter app. We have already tried "key" property of flutter Widget but after the build, it does not reflect the resource-id/accessibility-Id/content-desc based on the value given for the key of the element. 

To set the resourceId in native android app, please refer https://developer.android.com/guide/topics/resources/layout-resource. Similarly, I need resourceId in flutter app which can be recognized by AndroidDriver.

Additionally, We already opened a question on StackOverflow as well and awaiting for the solution if any.

**Comment by darshankawar** [comment]
@maulikpatel890 thanks for the details. Please see below related issues:

https://github.com/flutter/flutter/issues/17988
https://github.com/flutter/flutter/issues/18060
